### PR TITLE
Replace use of createHttpClient with mocks

### PR DESCRIPTION
**Pull request by cbracken**
package:http was recently removed from Flutter. This replaces the use of
createHttpClient, which was previously used to mock HTTP results. This
replaces that use with Mockito.

See: https://groups.google.com/d/msg/flutter-dev/AnqDqgQ6vus/spjSxiz6BgAJ

### CODE REVIEW

**Comment by Hixie** [comment]
@fkorotkov not clear why the cirrus jobs are failing here. Are they getting an old version of Flutter?

**Comment by fkorotkov** [comment]
Addressed in #39. Sorry for the inconvenience. 😅Forgot about this repo when added https://github.com/flutter/plugins/pull/415.

**Comment by fkorotkov** [comment]
#39 just got merged, so once this branch will be rebased, CI checks should pass. 👌

File: test/flutter_markdown_test.dart [hunk 1]
```diff
@@ -167,10 +168,10 @@ void main() {
       expect(tapResults, orderedEquals(['firstHref', 'secondHref']));
     });
   });
-  
+
   group('Images', () {
     setUpAll(() {
```
Comment (jonahwilliams) [inline test/flutter_markdown_test.dart]: `setUp` instead of `setUpAll`, since this only needs to run once and running it multiple times has no effect.

**Comment by jonahwilliams** [comment]
LGTM, besides nit

File: test/flutter_markdown_test.dart (hunk 1 above)
Comment (cbracken) [reply test/flutter_markdown_test.dart]: Done. Also added an explanatory comment.

**Pull request merged**